clang/lib/Driver/ToolChains/Arch/RISCV.cpp

```diff
@@ -239,6 +240,13 @@ StringRef riscv::getRISCVABI(const ArgList &Args, const llvm::Triple &Triple) {
   // rv64g | rv64*d -> lp64d
   // rv64* -> lp64
   StringRef Arch = getRISCVArch(Args, Triple);
+  if (Triple.getSubArch() == llvm::Triple::RISCV32SubArch_cheriot) {
+    llvm::Triple::OSType OS = Triple.getOS();
+    if (OS == llvm::Triple::CheriotRTOS)
+      return "cheriot";
+    else if (OS == llvm::Triple::UnknownOS)
+      return "cheriot-baremetal";
```

Comment:
What ABI are we getting for all of the ‘unknown’ things in the tests?

Reply:
Not totally sure I follow the question, but riscv32cheriot-unknown-unknown will imply "cheriot-baremetal" ABI. I think this best matches user expectations in that "i386-unknown-unknown" would be the target triple for bare metal i386, etc.